platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,237 @@
+| Broken link checking | True; will fail the build on the `master` branch only | True; will fail the build | False |
+| Accessibility | True; will fail the build | True; will fail the build | False |
+
+The Metalsmith script performs the check for broken internal
+links. This happens before the accessibility tests in the Jenkins
+build pipelines.
+
+When broken links are discovered during a full Jenkins build, we send
+a Slack notification with a link to the build. If Jenkins was building
+the production (`master`) branch, the build will also fail, to prevent
+broken links in production.
+
+Accessibility tests run the aXe checker on all pages found in the
+sitemap. Whenever the accessibility check fails, the Jenkins build
+fails. This happens after the content build and link checker.
+
+## Design
+I propose rearchitecting the `vets-website` build to separate the
+static content building into a process distinct from the website.
+
+![separate build diagram](images/separated-content-build-proposal.png)
+
+**Note:** Pulling the already-built content _can still_ have two
+different results, much like the current state, but since the content
+it's pulling will have already been validated, the likelihood of a
+failure is drastically reduced.
+
+### Changes
+- The `vets-website` build script will no longer build the content
+  - **Possible exceptions:**
+	- We may need to update some things in the
+	  pre-built content such as references to the new cache-busted
+	  filenames for JS and CSS assets
+	- It should still be responsible for building the pages for React
+	  apps, which would need to have the surrounding markup come from
+	  the templates
+- There will be a new repo for the content build process
+  - This will include scripts and templates
+- There will be a new Jenkins pipeline for building the content
+  - And everything that goes with this
+- There will be a (new?) S3 bucket which holds all the built content
+  - We currently do this with the cache, so this isn't much of a
+    deviation
+- The preview server will live in the new content-building repo
+- Sitemap accessibility tests will be done during the content build,
+  not the `vets-website` build
+- The `vets-website` Jenkins build will have the following changes
+  - There will be no cache fallback
+  - There will be no content caching
+  - There will be no sitemap accessibility testing
+  
+### Serving React apps
+`vets-website` should be responsible for making the pages where the
+React apps live, but not responsible for building the header, footer,
+and other surrounding markup. To make this possible, **I propose we
+have the content build produce an empty React app landing page
+template which the `vets-website` build script can use to make landing
+pages for each React application.** The `vets-website` build script
+would then update the landing page with references to the correct
+JavaScript bundle and CSS files to bootstrap the application.
```

Comment:
A question that's not necessarily related to this RFC specifically, but behavior may change with this method inadvertently: what do we do in the event that there's a conflict between content and react pages? i.e. someone creates a `/health/` page in Drupal, but there's already a `/health/` page in vets-website (or vice versa)

Reply:
Right, so currently, this isn't a thing we're handling at all. Drupal pages are set to override both React landing pages and `vagov-content` pages. We'd probably want to throw up a warning to some place visible (Slack?) and use the React landing page. It would then be on a case-by-case basis what the resolution looks like.